src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newStates = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        const StateBfAdd = { ...(newStates[newStates.length - 1] || state) };
```

Comment:
[NAMING] - The variable name 'StateBfAdd' does not follow proper naming conventions. Consider using 'stateBeforeAdd' or 'stateCopy' to improve readability.